src/clist.rs

```diff
@@ -475,6 +714,46 @@ mod tests {
         assert_eq!(clist.len(), 0);
     }
 
+    #[test]
+    fn test_is_empty() {
+       let mut clist = CList::with_capacity(2);
+       assert_eq!(clist.is_empty(), true);
+
+       let a_addr = clist.insert_ref('a', ROOT_ADDR, true);
+       assert_eq!(clist.is_empty(), false);
+
+       clist.remove(a_addr);
+       assert_eq!(clist.is_empty(), true);
+
+    }
+
+    #[should_panic]
+    #[test]
+    fn test_assert_not_free(){
+        let mut clist = CList::with_capacity(1);
+        let a_addr = clist.insert_ref('a', ROOT_ADDR, true);
+        clist.remove(a_addr);
+        clist.assert_not_free(&a_addr);
+    }
+
+    #[test]
+    fn test_get() {
+        let mut clist = CList::with_capacity(1);
+        let a_addr = clist.insert_ref('a', ROOT_ADDR, true);
+        assert_eq!(*clist.get(a_addr), 'a');
+        assert_ne!(*clist.get(a_addr), 'b');
+    }
+
+    #[test]
+    fn test_get_mut() {
+        let mut clist = CList::with_capacity(1);
+        let _a_addr = clist.insert_ref('a', ROOT_ADDR, true);
+        let b_addr = clist.insert_ref('b', ROOT_ADDR, true);
+        
+        assert_eq!(*clist.get(b_addr), 'b');
+        assert_ne!(*clist.get(b_addr), 'a');
+    }
```

Comment:
Here you used `get` instead of `get_mut`

But the idea behind `get_mut` is for mutability.

Something like
```rust
let chr = clist.get_mut(a_addr);
*chr = 'z';
```
And then assert that the list now contains `'z'` instead of `'a'`.

Also, `assert_ne` is not necessary